src/webgpu/util/math.ts

```diff
@@ -683,38 +701,44 @@ export function sparseF32Range(): Array<number> {
  * vector to get a spread of testing over the entire range. This reduces the
  * number of cases being run substantially, but maintains coverage.
  */
-export const kVectorTestValues = {
-  2: sparseF32Range().flatMap(f => [
-    [f, 1.0],
-    [1.0, f],
-    [f, -1.0],
-    [-1.0, f],
-  ]),
-  3: sparseF32Range().flatMap(f => [
-    [f, 1.0, 2.0],
-    [1.0, f, 2.0],
-    [1.0, 2.0, f],
-    [f, -1.0, -2.0],
-    [-1.0, f, -2.0],
-    [-1.0, -2.0, f],
-  ]),
-  4: sparseF32Range().flatMap(f => [
-    [f, 1.0, 2.0, 3.0],
-    [1.0, f, 2.0, 3.0],
-    [1.0, 2.0, f, 3.0],
-    [1.0, 2.0, 3.0, f],
-    [f, -1.0, -2.0, -3.0],
-    [-1.0, f, -2.0, -3.0],
-    [-1.0, -2.0, f, -3.0],
-    [-1.0, -2.0, -3.0, f],
-  ]),
-};
+export function vectorTestValues(dimension: number, source: String): number[][] {
```

Comment:
I'm really not keen on the propagation of the `source` as a string. It's pushing details of one particular part of tests into core maths libraries.

This isn't something you've introduced, but something that's being cemented. 

This is about finite vs non-finite test cases, right? So let's call this what it is. Something like `finite_only: boolean`.

Reply:
ACK and done.